src/main/java/com/google/googleidentity/oauth2/authorizationcode/AuthorizationCodeService.java

```diff
@@ -72,21 +89,29 @@ public String getCodeForRequest(OAuth2Request request){
         return codeStore.consumeCode(code);
     }
 
-    private String generateCode(){
+    /**
+     * Associate with client and username to reduce collisions.
+     *
+     * @param clientID
+     * @param username
+     * @return
+     */
+    private String generateCode(String clientID, String username){
+        byte[] fixInfo =
```

Comment:
nit: fixinfo is not a very descriptive variable name, and it's not immediately clear what's being fixed if you don't know what was broken before.

I would recommend names that are more descriptive of what the variables mean. It will make it easier to read.

fixInfo -> prefixBytes
bytes -> authCodeBytes
fixLength -> numPrefixBytesTocopy